File: src/com/Torvald/Terrarum/GameItem/ItemCodex.java
package com.Torvald.Terrarum.GameItem;

import java.util.HashMap;

/**
 * Created by minjaesong on 16-03-11.
 */
public class ItemCodex {

    /**
     * &lt;ItemID or RefID for Actor, TheItem&gt;
     * Will return corresponding Actor if ID >= 32768
     */
    private static HashMap<Long, InventoryItem> itemCodex;

    public static InventoryItem getItem(long code) {
        return itemCodex.get(code);
    }

}

File: src/com/Torvald/Terrarum/GameItem/ItemCodex.kt
package com.Torvald.Terrarum.GameItem

import java.util.*

/**
 * Created by minjaesong on 16-03-15.
 */
object ItemCodex {
    /**
     * &lt;ItemID or RefID for Actor, TheItem&gt;
     * Will return corresponding Actor if ID >= 32768
     */
    private val itemCodex: HashMap<Long, InventoryItem> = HashMap(
            // hashmap init here
    )


    fun getItem(code: Long): InventoryItem {
        return itemCodex[code]!!
    }
}